service/matching/matching_engine.go

```diff
@@ -1372,34 +1378,102 @@ func (e *matchingEngineImpl) unloadTaskQueuePartition(unloadPM taskQueuePartitio
 	delete(e.partitions, key)
 	e.partitionsLock.Unlock()
 	foundTQM.Stop()
-	e.updateTaskQueueGauge(foundTQM, false, -1)
+	e.updateTaskQueuePartitionGauge(unloadPM, -1)
+}
+
+// Responsible for emitting and updating loaded_physical_task_queue_count metric
+func (e *matchingEngineImpl) updatePhysicalTaskQueueGauge(pm taskQueuePartitionManager, versioned string, delta int) {
+	// assuming version contains one of: “unversioned” or "buildId” or “versionSet”
```

Comment:
better to just get the physical queue and calculate `versioned` value in side here. Keeping the metrics concerns private.
```suggestion
func (e *matchingEngineImpl) updatePhysicalTaskQueueGauge(pq pysicalTaskQueuePartitionManagerImpl, delta int) {
```